parcels/field.py

```diff
@@ -1785,19 +1852,23 @@ def data_access(self):
             data = np.array(data)
         else:
             if isinstance(data, da.core.Array):
-                if self.field_chunksize == 'auto' and data.shape[-2:] == data.chunksize[-2:]:
-                    pass
-                elif self.field_chunksize == 'auto' and self.rechunk_callback_fields is not None:
+                if self.field_chunksize == 'auto' and data.shape[-2:] == data.chunksize[-2:] and not self.chunking_finalized:
+                    self.chunking_finalized = True
+                elif self.field_chunksize == 'auto' and self.rechunk_callback_fields is not None and not self.chunking_finalized:
                     data = data.rechunk(self.field_chunksize)
                     self.chunk_mapping = {}
                     chunkIndex = 0
                     for chunkDim in data.numblocks[1:]:
                         self.chunk_mapping[chunkIndex] = chunkDim
                         chunkIndex += 1
+                    self._chunkmap_to_chunksize()
                     self.rechunk_callback_fields()
-                elif self.field_chunksize != 'auto':
+                    self.chunking_finalized = True
+                elif self.field_chunksize != 'auto' and not self.chunking_finalized:    # why ? why here ? why at every single data access, instead of fixing that size when you open the file ?!
```

Comment:
This is a somewhat desperate-sounding comment. As if someone who didn't write the code added it here? Could you expand?

Reply:
oops - old left-over from my private repo - will be removed (what is standing there actually now fixes what I was mourning over at that point).